app/org/sagebionetworks/bridge/validators/SurveySaveValidator.java

```diff
@@ -143,77 +144,115 @@ private void validateRules(Errors errors, List<SurveyElement> elements) {
+                SurveyRule rule = rules.get(j);
+                if (rule.getOperator() != SurveyRule.Operator.ALWAYS) {
+                    errors.pushNestedPath(propertyPath);
+                    errors.rejectValue(fieldName+"["+j+"]", "only valid with the 'always' operator");
+                    errors.popNestedPath();
+                }
+            }
+        }        
+    }
     
-    private void validateSkipToTargetExists(Errors errors, SurveyRule rule, Set<String> alreadySeenIdentifiers,
-            String propertyPath, String fieldPath) {
-        
-        if (rule.getSkipToTarget() != null) {
-            if (!alreadySeenIdentifiers.contains(rule.getSkipToTarget())) {
+    private void validateOneRuleSet(Errors errors, List<SurveyRule> rules, Set<String> alreadySeenIdentifiers,
+            String propertyPath, String fieldName) {
+        if (rules != null) {
+            for (int j=0; j < rules.size(); j++) {
+                SurveyRule rule = rules.get(j);
+                String fieldPath = fieldName+"["+j+"]";
+                
                 errors.pushNestedPath(propertyPath);
-                errors.rejectValue(fieldPath, "has a skipTo identifier that doesn't exist: " + rule.getSkipToTarget());
+                
+                if (!hasOneAction(rule)) {
+                    errors.rejectValue(fieldPath, "must have one and only one action");
+                }
+                if (assignedDataGroupDoesNotExist(rule)) {
+                    errors.rejectValue(fieldPath, "has a data group '" + rule.getAssignDataGroup()
+                            + "' that is not a valid data group: " + COMMA_SPACE_JOINER.join(dataGroups));            
+                }
+                if (afterRuleControlsDisplay(fieldPath, rule)) {
+                    errors.rejectValue(fieldPath, "specifies display after screen has been shown");
+                }
+                if (skipToBackReferencesQuestion(alreadySeenIdentifiers, rule)) {
+                    errors.rejectValue(fieldPath, "back references question " + rule.getSkipToTarget());
+                }
+                // Split rules by their operator, the operators determines what data is being tested. 
+                // Only validate fields that are relevant for the operator, ignore the other.
+                if (SurveyRule.SET_OPERATORS.contains(rule.getOperator())) {
+                    // tests against data groups
+                    if (isEmpty(rule.getDataGroups())) {
+                        errors.rejectValue(fieldPath, "should define one or more data groups");
+                    } else if (!dataGroups.containsAll(rule.getDataGroups())) {
+                        errors.rejectValue(fieldPath,
+                            "contains dataGroups '" + COMMA_SPACE_JOINER.join(rule.getDataGroups())
+                                    + "' that are not valid data groups: " + COMMA_SPACE_JOINER.join(dataGroups));
+                    }
```

Comment:
Unit tests for these?
* empty data groups for ANY
* invalid data groups for ANY
* at least one invalid rule with ALL to verify that that's being tested